### PR TITLE
Support search redirects to PennyLane.ai

### PR DESCRIPTION
**Pull request by Mandrenkov**
**Context:**

Search in Sphinx is, both functionally and aesthetically, lacking. This PR introduces a theme option (for the [PennyLane Sphinx Theme](https://github.com/PennyLaneAI/pennylane-sphinx-theme)) to redirect users from `search.html` to https://pennylane.ai/search (with the right query parameters).

It's worth noting that PennyLane Docs are not surfaced yet in PennyLane.ai search results but this will change soon.

**Description of the Change:**

1. Added a theme option, `search_on_pennylane_ai`, which redirects users from `search.html` (i.e., after submitting a query using the search box in the LHS ToC) to https://pennylane.ai/search with the following query parameters:
    * `contentType`: Set to `DOC`.
    * `q`: Set to the query in the search box.
    * `version`: Set to `stable`, unless `/latest/` is part of the current URL, in which case `latest` is used instead.

**Benefits:**

1. PennyLane Docs users have easier access to a better search experience.

**Possible Drawbacks:**

_None._

**Related GitHub Issues:**

_None._

---

**Verification:**

See https://xanaduai-pennylane--67.com.readthedocs.build/projects/sphinx-theme/en/67/.

**Merge Checklist:**
* [X] PennyLane Docs are surfaced in PennyLane.ai search results.

### CODE REVIEW

File: xanadu_sphinx_theme/search.html [hunk 1]
```diff
@@ -1,3 +1,18 @@
+{% if theme_search_on_pennylane_ai %}
+  <!-- If JavaScript is enabled, redirect users to https://pennylane.ai/search. -->
+  <script type="text/javascript">
+    const search_version = window.location.pathname.includes("/latest/") ? "latest" : "stable";
+    const search_query = new URLSearchParams(window.location.search).get("q")
```
Comment (AndrewGardhouse) [inline xanadu_sphinx_theme/search.html:5]:
```suggestion
    const search_query = new URLSearchParams(window.location.search).get("q") || "";
```

File: xanadu_sphinx_theme/search.html [hunk 2]
```diff
@@ -1,3 +1,18 @@
+{% if theme_search_on_pennylane_ai %}
+  <!-- If JavaScript is enabled, redirect users to https://pennylane.ai/search. -->
```
Comment (anthayes92) [inline xanadu_sphinx_theme/search.html:2]: > If JavaScript is enabled

Is this equivalent to `theme_search_on_pennylane_ai = True`? Or is there another way JavaScript can be enabled/disabled?

**Review by anthayes92 — APPROVED** [review]
Nice addition! :rocket:

File: xanadu_sphinx_theme/search.html (hunk 2 above)
Comment (Mandrenkov) [reply xanadu_sphinx_theme/search.html:2]: Not quite. This is usually something you configure in your browser (e.g., via an extension). Additionally, if you try to fetch a webpage using something like `wget` or cURL, no JavaScript is executed.

**Review by AndrewGardhouse — APPROVED** [review]
LGTM!

**Pull request merged**